engine/src/main/java/org/hibernate/validator/internal/properties/Callable.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.properties;
+
+import java.lang.reflect.Type;
+
+import org.hibernate.validator.internal.util.ExecutableHelper;
+import org.hibernate.validator.internal.util.ExecutableParameterNameProvider;
+
+/**
+ * @author Marko Bekhta
+ * @author Guillaume Smet
+ */
+public interface Callable extends Constrainable {
+
+	boolean hasReturnValue();
+
+	boolean hasParameters();
+
+	int getParameterCount();
+
+	Type getParameterGenericType(int index);
+
+	Class<?>[] getParameterTypes();
+
+	String getParameterName(ExecutableParameterNameProvider parameterNameProvider, int parameterIndex);
+
+	boolean isPrivate();
+
+	String getSignature();
```

Comment:
I suppose that's for logging only?

Reply:
Not only. The signature is used a mean to identify the executable in various parts of the code.